plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/util/OpenShiftResourceUniqueId.java

```diff
@@ -61,14 +63,34 @@ public static String getResourceName(String uniqueId) {
 		if (StringUtils.isEmpty(uniqueId)) {
 			return null;
 		}
-		int index = uniqueId.indexOf(UNIQUE_ID_PROJECT_NAME_DELIMITER);
-		if (index == -1) {
-			return null;
-		}
-		return uniqueId.substring(index + 1);
+        String[] comps = uniqueId.split(String.valueOf(UNIQUE_ID_PROJECT_NAME_DELIMITER));
+        if (comps.length > 0) {
+            return comps[comps.length - 1];
+        } else {
+            return null;
+        }
 	}
 
 	/**
+	 * Return the OpenShift resource kind associated with this resource id. If an old
+	 * resource id is given, then Service is assumed.
+	 * 
+	 * @param uniqueId the OpenShift resource id
+	 * @return the kind of the OpenShift resource
+	 */
+    public static String getKind(String uniqueId) {
+        if (StringUtils.isEmpty(uniqueId)) {
+            return null;
+        }
+        String[] comps = uniqueId.split(String.valueOf(UNIQUE_ID_PROJECT_NAME_DELIMITER));
+        if (comps.length == 2) {
+            return ResourceKind.SERVICE;
+        } else {
+            return comps[1];
```

Comment:
this should only be performed once we checked that there are actually >= 2 elements in comps. We otherwise potentially run into an ArrayIndexOutOfBoundsException.